src/components/markdown-renderer/basic-markdown-renderer.tsx

```diff
@@ -0,0 +1,112 @@
+import MarkdownIt from 'markdown-it'
+import abbreviation from 'markdown-it-abbr'
+import definitionList from 'markdown-it-deflist'
+import emoji from 'markdown-it-emoji'
+import footnote from 'markdown-it-footnote'
+import imsize from 'markdown-it-imsize'
+import inserted from 'markdown-it-ins'
+import marked from 'markdown-it-mark'
+import subscript from 'markdown-it-sub'
+import superscript from 'markdown-it-sup'
+import React, { ReactElement, RefObject, useMemo, useRef } from 'react'
+import { Alert } from 'react-bootstrap'
+import ReactHtmlParser from 'react-html-parser'
+import { Trans } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import { ApplicationState } from '../../redux'
+import { ShowIf } from '../common/show-if/show-if'
+import './markdown-renderer.scss'
+import { combinedEmojiData } from './markdown-it-plugins/emoji/mapping'
+import { linkifyExtra } from './markdown-it-plugins/linkify-extra'
+import { MarkdownItParserDebugger } from './markdown-it-plugins/parser-debugger'
+import { ComponentReplacer } from './replace-components/ComponentReplacer'
+import { buildTransformer } from './utils/html-react-transformer'
+import { calculateNewLineNumberMapping } from './utils/line-number-mapping'
+import { LineKeys } from './utils/types'
+
+export interface AdditionalMarkdownRendererProps {
```

Comment:
Maybe this should be moved into a separate type definition file?

Reply:
well true. I moved it into the types.d.ts